SFM and DisparityModule are not working as expected

Today me and @Nicogene, tried to use both `SFM` and `DisparityModule` on the iCubGenova01 setup (Ubuntu 18.04, OpenCV 3.4.0 build with CUDA 10.0).

Unfortunately, we were not able to run them as expected (our test consisted in launching the module and check the output disparity from port `/DisparityModule/disp:o` or `/SFM/disp:o` depending on the module).

`DisparityModule` runs but, after connecting the input images, it does not produce any output in `/DisparityModule/disp:o`.

`SFM`, instead, stops at runtime with the following error:

```
OpenCV Error: Bad argument (unrecognized or unsuported array type) in cvPtr2D, file (...)
terminate called after throwing an instance of 'cv::Exception'
what(): /usr/loca/src/robot/opencv/modules/core/src/array.cpp:1839: error: (-5) unrecognized or unsupported array type in function cvPtr2D
```

A working commit that we found, for SFM, is https://github.com/robotology/stereo-vision/commit/9cbc03ae5d7983bc908d2089abfb5cfae5e55f42. We tried to bisect from there but we were not able to compile halfways commits.

--- a/modules/DisparityModule/DispModule.cpp
+++ b/modules/DisparityModule/DispModule.cpp
@@ -1604,24 +1604,16 @@ int main(int argc, char *argv[])
     // rough quick check of the command line parameters, 
     // has to be definitely made in a smart way
 
-    bool use640 = false;
-
-    for(int i = 0; i < argc; i++)
-        if(strcmp(argv[i], "--use640") == 0)
-        {
-            use640 = true;
-            break;
-        }
-
     ResourceFinder rf;
+    rf.setDefaultContext("cameraCalibration");
+    rf.configure(argc, argv);
 
-    if(use640)
+    if(rf.check("use640"))
         rf.setDefaultConfigFile("icubEyes_640x480.ini");
     else
         rf.setDefaultConfigFile("icubEyes.ini");
 
-    rf.setDefaultContext("cameraCalibration");
-    rf.configure(argc,argv);
+    rf.configure(argc, argv);
 
     DispModule mod;
     

--- a/modules/DisparityModule/DispModule.cpp
+++ b/modules/DisparityModule/DispModule.cpp
@@ -386,6 +386,10 @@ void DispModule::initializeStereoParams()
     this->stereo_parameters.wls_lambda = 4500.;
     this->stereo_parameters.wls_sigma = 1.0;
 
+    this->stereo_parameters.stereo_matching = SM_MATCHING_ALG::SGBM_OPENCV;
+    this->stereo_parameters.BLFfiltering = SM_BLF_FILTER::BLF_DISABLED;
+    this->stereo_parameters.WLSfiltering = SM_WLS_FILTER::WLS_DISABLED;
+
     this->original_parameters = this->stereo_parameters;
 }
 

